### PR TITLE
feat: added pmap read me

### LINKED ISSUE #143: Create PMAP readme doc
### TL;DR

Create PMAP readme doc including how to use pmap and common issues encountered in pmap

### Detailed design

_No response_

### Alternatives considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by capri-xiyue**
fixed #143

### CODE REVIEW

**Review by sqin2019 — APPROVED** [review]

File: docs/example/wipeout_plan.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+plan_id: 'retention/product_team_a/product_x_user_data_retention'
+title: 'Wipeout Plan for User Data of Product X'
+description: 'This is an example wipeout plan.'
+scope: 'Product X'
+owner: 'producscopt_team_a@google.com'
```
Comment (yolocs) [inline docs/example/wipeout_plan.yaml:5]:
```suggestion
owner: 'producscopt_team_a@example.com'
```

File: docs/example/wipeout_plan.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+context: 'PDD link address'
```
Comment (yolocs) [inline docs/example/wipeout_plan.yaml:6]: PDD doesn't make sense here

File: docs/example/wipeout_plan.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      action: 'Deletion'
+      purpose: 'Spam and Abuse'
+      trigger:
+        trigger_event: 'Identifier expiration'
+        delay: '356 days'
+justification: 'Store data for expired users marked as abusive.'
+exception_information:
+  controls: 'Controls are...'
+  related_documents_or_issue: 'Issue link address'
+last_modified: '2023/06/13'
```
Comment (yolocs) [inline docs/example/wipeout_plan.yaml:27]: Can we skip non-essential fields?

File: README.md [hunk 4]
```diff
@@ -3,3 +3,196 @@
 **This is not an official Google product.**
 
 Privacy data mapping and related plans management.
+
+## Set Up
```
Comment (yolocs) [inline README.md:7]: Have architecture and end user workflow before the set up.

File: README.md [hunk 5]
```diff
@@ -3,3 +3,196 @@
...
+
+    -  Service account used in Authenticating via Workload Identity Federation
+       needs [roles/storage.objectCreator] 
+       to snapshot the privacy mapping data/retention plans from GitHub to GCS.
+
+### GitHub Central Repository
+
+The GitHub central repository is the source of truth
+for privacy data mappings/wipeout plans.
+We rely on GitHub to preserve change history and enable multi-person review.
```
Comment (yolocs) [inline README.md:25]: Some background knowledge would be very helpful here. E.g. why data needs to be mapped. I can help write that if needed.

File: README.md [hunk 6]
```diff
@@ -3,3 +3,196 @@
...
+
+The central privacy team can determine how to group privacy data mappings/wipeout plans as long as
+at least one level of group are needed
+(Sub folders in the root of the central GitHub repository are needed).
+Files containing privacy data mappings/wipeout plans can’t be stored
+directly in the root of the central GitHub repository.
+
+#### Privacy Data Mapping
+
+* Presubmit workflows for sanity checks
```
Comment (yolocs) [inline README.md:35]: I wonder if it's better to save the workflows as separate yaml files.

**Review by yolocs — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (sailorlqh) [reply README.md:7]: I am working on the architecture, and will add that later in a different PR.

File: docs/example/wipeout_plan.yaml (hunk 1 above)
Comment (capri-xiyue) [reply docs/example/wipeout_plan.yaml:5]: done

File: docs/example/wipeout_plan.yaml (hunk 3 above)
Comment (capri-xiyue) [reply docs/example/wipeout_plan.yaml:27]: done

File: README.md (hunk 4 above)
Comment (capri-xiyue) [reply README.md:7]: Discussed offline. I will use Qinhang's architecture graph in this PR to make it easier to review. Thanks to @sailorlqh for drawing the archiecture graph

File: README.md (hunk 6 above)
Comment (capri-xiyue) [reply README.md:35]: done

File: README.md [hunk 7]
```diff
@@ -2,4 +2,149 @@
...
 **This is not an official Google product.**
 
-Privacy data mapping and related plans management.
+## Background
+
+Privacy data management is the process of collecting, storing, using, and
+disposing of personal data in a way that protects the privacy of individuals. It
+is a critical part of any organization that collects or uses personal data, and
+organizations are typically required to maintain compliance with policies set by
+regulatory bodies.
```
Comment (yolocs) [inline README.md:11]:
```suggestion
Privacy data management is the process of collecting, storing, using, and
disposing of data in a way that protects the privacy of users. It
is a critical part of any organization that collects or uses user data, and
organizations are typically required to maintain compliance with policies set by
regulatory bodies.
```

File: README.md [hunk 8]
```diff
@@ -2,4 +2,149 @@
...
+* Which privacy policy applies to
+the datastore that stores user data(aka Data Mapping). This includes
+understanding how the organization's privacy policies are applied to different
+systems and databases. 
+* The visibility of the privacy compliance. This includes
+being able to track and monitor the organization's compliance with its privacy
+policies and applicable laws and regulations.
+
+PMAP provides a solution for the first three problems. We are working on a
+solution to provide visibility of privacy compliance in the near future.
```
Comment (yolocs) [inline README.md:32]:
```suggestion
To ensure that organizations maintain compliance with policies set by regulatory
bodies, they need to know the following:

* **The requirements for what teams
must do, driven by legal requirements or external commitments (aka. privacy
policy and compliance controls). This includes translating the comprehensive
external legal requirements into requirements that are tailored to products and
services of the organizations. 
* **Where the user data is stored or processed.** This includes understanding the
different systems and databases that store or process user data, as well as the
physical locations where user data is stored.
* **Which privacy policy or compliance control applies to
the system that stores/processes user data (aka. data mapping).** This includes
understanding how the organization's privacy policies and controls are applied to
different systems and databases. 
* **The visibility of the privacy compliance.** This includes
being able to track and monitor the organization's compliance with its privacy
policies and applicable laws and regulations.

PMAP provides a solution for the first three problems. We are working on a
solution to provide visibility of privacy compliance in the near future.
```

Comment (yolocs) [reply README.md:32]: btw, this part doesn't seem properly formatted. You need an empty line before bullet points.

File: README.md [hunk 9]
```diff
@@ -2,4 +2,149 @@
...
+
+*   Registration - Data owners and policy owners will register data mappings and
+    privacy policies in a central GitHub repository.
+*   GCS Snapshots - Snapshot the data mappings and privacy policies from GitHub
+    to GCS with Workload Identity Federation.
+*   Additional Processors - Extension point for validation and enrichment.
+*   Processing Service - The service that is responsible for ingesting,
+    validating and storing the data mappings and privacy policies.
+*   Storage and Analysis - The data warehouse for processed data mappings and
+    privacy policies, and UI for dashboarding.
```
Comment (yolocs) [inline README.md:46]:
```suggestion
*   **Registration** - Data owners and policy owners will register data mappings and
    privacy policies in a central GitHub repository.
*   **GCS Snapshots** - Snapshot the data mappings and privacy policies from GitHub
    to GCS with Workload Identity Federation.
*   **Additional Processors** - Extension point for validation and enrichment.
*   **Processing Service** - The service that is responsible for ingesting,
    validating and storing the data mappings and privacy policies.
*   **Storage and Analysis** - The data warehouse for processed data mappings and
    privacy policies, and UI for dashboarding.
```

File: README.md [hunk 10]
```diff
@@ -2,4 +2,149 @@
...
+
+### Why GitHub
+
+We choose GitHub as it can preserve change history and enable multi-person
+review and approval. Change history and review&approval process are crucial in
```
Comment (yolocs) [inline README.md:51]:
```suggestion
review and approval. Change history and review/approval process are crucial in
```

File: README.md [hunk 11]
```diff
@@ -2,4 +2,149 @@
...
+privacy data management.
+
+### Why BigQuery
+
+We choose BigQuery for its excellent analytics support: 
+* Be able to visualize
+data to reveal meaningful insights 
+* Ideally be able to join data from other
+data sources in the future to achieve the privacy compliance monitoring.
```
Comment (yolocs) [inline README.md:60]: Again, you need an empty line before bullet points to have the right formatting

```suggestion
We choose BigQuery for its excellent analytics support: 

* Be able to visualize
data to reveal meaningful insights 
* Be able to join data from other
data sources in the future to achieve the privacy compliance monitoring.
```

File: README.md [hunk 12]
```diff
@@ -2,4 +2,149 @@
...
+```
+-  Service account used in Authenticating via Workload Identity Federation
+   needs [roles/storage.objectCreator]
+   to snapshot the privacy mapping data/retention plans from GitHub to GCS.
+```
+
+### GitHub Central Repository
+
+GitHub central repository is the source of truth for data mappings and privacy
+policies.
```
Comment (yolocs) [inline README.md:82]:
```suggestion
GitHub central repository is the source of truth for data mappings and policies/controls.
```

File: README.md [hunk 13]
```diff
@@ -2,4 +2,149 @@
...
+
+We choose BigQuery for its excellent analytics support: 
+* Be able to visualize
+data to reveal meaningful insights 
+* Ideally be able to join data from other
+data sources in the future to achieve the privacy compliance monitoring.
+
+## Set Up
+
+The central privacy team need to complete the steps below.
```
Comment (yolocs) [inline README.md:64]:
```suggestion
The central privacy/compliance eng team need to complete the steps below.
```

File: README.md [hunk 14]
```diff
@@ -2,4 +2,149 @@
...
+### GitHub Central Repository
+
+GitHub central repository is the source of truth for data mappings and privacy
+policies.
+
+The central privacy team can determine how to group privacy data
+mappings/wipeout plans as long as at least one level of group are needed (Sub
+folders in the root of the central GitHub repository are needed). Files
+containing privacy data mappings/wipeout plans can’t be stored directly in the
+root of the central GitHub repository.
```
Comment (yolocs) [inline README.md:88]:
```suggestion
The central privacy/compliance eng team can determine how to group data
mappings and policies/controls as long as at least one level of group are needed (sub
folders in the root of the central GitHub repository are needed). Files
containing the data mappings or policies/controls can’t be stored directly in the
root of the central GitHub repository.
```

File: README.md [hunk 15]
```diff
@@ -2,4 +2,149 @@
...
+    [here](./docs/example/mapping_data_validation.yaml).
+
+*   Postsubmit workflows to snapshot added_files and modified_files of privacy
+    data mappings to GCS, see example
+    [here](./docs/example/snapshot_mapping_data_change.yaml).
+
+*   Cron Workflows to snapshot the all files of privacy data mappings to GCS,
+    see example [here](./docs/example/snapshot_mapping_data_copy.yaml).
+
+#### Privacy Policy
```
Comment (yolocs) [inline README.md:102]:
```suggestion
#### Policy and Control
```

File: README.md [hunk 16]
```diff
@@ -2,4 +2,149 @@
...
+GitHub central repository is the source of truth for data mappings and privacy
+policies.
+
+The central privacy team can determine how to group privacy data
+mappings/wipeout plans as long as at least one level of group are needed (Sub
+folders in the root of the central GitHub repository are needed). Files
+containing privacy data mappings/wipeout plans can’t be stored directly in the
+root of the central GitHub repository.
+
+#### Privacy Data Mapping
```
Comment (yolocs) [inline README.md:90]:
```suggestion
#### Data Mapping
```

File: README.md [hunk 17]
```diff
@@ -2,4 +2,149 @@
...
+  pmap_container_image             = "us-docker.pkg.dev/abcxyz-artifacts/docker-images/pmap:0.0.3-amd64"
+  bigquery_table_delete_protection = true
+  # This is used when searching global Cloud Resources like GCS bucket.
+  pmap_specific_envvars            = { "PMAP_MAPPING_DEFAULT_RESOURCE_SCOPE" : "YOUR_DEFAULT_RESOURCE_SCOPE" }
+}
+```
+
+## End User Workflows
+
+### Policy Owner
```
Comment (yolocs) [inline README.md:133]:
```suggestion
### Policy/Control Owner
```

File: README.md [hunk 18]
```diff
@@ -2,4 +2,149 @@
...
+
+*   Create a privacy policy (wipeout plan etc.) by opening a PR in the yaml
+    format file (file has `.yaml` filename suffix) under the sub folder where
+    stores
+    all the privacy policies. See example
+    [here](./docs/example/wipeout_plan.yaml).
```
Comment (yolocs) [inline README.md:139]:
```suggestion
*   Create a policy/control (e.g. a wipeout plan) by opening a PR and add a `yaml`
    file under the sub folder where
    stores
    all the policies/controls. See example
    [here](./docs/example/wipeout_plan.yaml).
```

File: README.md [hunk 19]
```diff
@@ -2,4 +2,149 @@
...
+
+### Data Owner
+
+*   Register and annotate resources to associate the resources to its specific
+    wipeout plan by opening a PR in the yaml format file (file has `.yaml`
+    filename suffix) under the sub folder where stores
+    all the privacy data mappings. See example
+    [here](./docs/example/resource_mapping.yaml). **NOTE:** The association of
+    the resource to the wipeout plan is achieved via `annotations` field.
```
Comment (yolocs) [inline README.md:148]:
```suggestion
*   Register and annotate resources to associate the resources to its specific
    policies/controls by opening a PR and add a mapping `yaml` file under the sub folder where stores
    all the data mappings. The association of
    the resource to the corresponding policies/controls is achieved via `annotations` field.
    See example
    [here](./docs/example/resource_mapping.yaml).
```

File: docs/example/resource_mapping.yaml
Comment (yolocs) [inline docs/example/resource_mapping.yaml:1]: mapping / policy examples should be in a separate folder from the workflows

**Review by yolocs — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (capri-xiyue) [reply README.md:11]: done

File: docs/example/resource_mapping.yaml
Comment (capri-xiyue) [reply docs/example/resource_mapping.yaml:1]: done

**Comment by capri-xiyue** [comment]
Thanks to @sailorlqh for drawing the architecture graph!

**Review by yolocs — APPROVED** [review]

**Pull request merged**